contracts/ssnlist.scilla

```diff
@@ -767,19 +773,29 @@ procedure CalcStakeDelegPerCycle(tmp_arg: TmpArg)
     buf_opt <- buff_deposit_deleg[deleg][ssn_operator][last2_reward_cycle];
     comb_opt = option_add cur_opt buf_opt;
 
+    last_amt_o <- deleg_stake_per_cycle[deleg][ssn_operator][last_reward_cycle];
+    last_amt = option_uint128_value last_amt_o;
+    staking_and_rewards_per_cycle_for_ssn_opt <- stake_ssn_per_cycle[ssn_operator][reward_cycle];
+    (* the rewards for this cycle also computed within this procedure, so we no long keep histric data *)
+    delete deleg_stake_per_cycle[deleg][ssn_operator][last_reward_cycle];
     delete direct_deposit_deleg[deleg][ssn_operator][last_reward_cycle];
     delete buff_deposit_deleg[deleg][ssn_operator][last2_reward_cycle];
 
-    last_amt_o <- deleg_stake_per_cycle[deleg][ssn_operator][last_reward_cycle];
-    last_amt = option_uint128_value last_amt_o;
+    staking_of_deleg = match comb_opt with
+    | Some stake => builtin add last_amt stake
+    | None => last_amt
+    end;
+    deleg_stake_per_cycle[deleg][ssn_operator][reward_cycle] := staking_of_deleg;
     
-    match comb_opt with
-    | Some stake =>
-      total_stake = builtin add last_amt stake;
-      deleg_stake_per_cycle[deleg][ssn_operator][reward_cycle] := total_stake
+    
+    match staking_and_rewards_per_cycle_for_ssn_opt with
+    | Some (SSNCycleInfo total_staking total_rewards) =>
+      reward_tmp = builtin mul total_rewards staking_of_deleg;
+      reward = builtin div reward_tmp total_staking;
+      IncreaseReward ssn_operator deleg reward
     | None =>
```

Comment:
Curious to know in what case this will be `None`?

Reply:
In the case ssn performed too bad and verifier didn't reward it in a specific cycle.